src/ContextMenu.js

```diff
@@ -124,7 +134,11 @@ export default class ContextMenu extends AbstractMenu {
     handleHide = (e) => {
         if (this.state.isVisible && (!e.detail || !e.detail.id || e.detail.id === this.props.id)) {
             this.unregisterHandlers();
-            this.setState({ isVisible: false, selectedItem: null, forceSubMenuOpen: false });
+            flushSync(() => {
```

Comment:
We get the issue described in https://github.com/reactwg/react-18/discussions/21 in the part `Does this break anything for Classes?`.

Indeed we rely on the new state in `handleShow` above. I couldn't find another solution than using `flushSync` here but I'll admit I didn't think this a lot :-) Happy to try other solutions if you can think of any.

Reply:
Hm, yeah, I couldn't think of a better solution off the top of my head either. I can't say that I like the `ReactDOM.flushSync` overall, but well, it's ok to keep it for this I think.